scheduler.py

```diff
@@ -1,178 +0,0 @@
-    ticker_count = {}
-    hot_posts = reddit.subreddit('wallstreetbets').hot(limit=500)
-
-    for post in hot_posts:
-        word_in_title = post.title.split()
-        add_to_dict(word_in_title, ticker_dict, ticker_count)
-        word_in_content = post.selftext.split()
-        add_to_dict(word_in_content, ticker_dict, ticker_count)
-
-    sorted_ticker_count = dict(
-        sorted(ticker_count.items(), key=lambda item: item[1], reverse=True)[:TOTAL_NUMBER_OF_STOCKS])
-
-    return sorted_ticker_count
-
-
-# Helper function to aggregate the count of each ticker in ticker_count
-# Filters out tickers in the blacklist
-def add_to_dict(word_in_post, ticker_dict, ticker_count):
-    for word in word_in_post:
-        clean_word = word.strip()
-        if clean_word in TICKER_BLACKLIST:
-            continue
-        if clean_word in ticker_dict:
-            ticker_count[clean_word] = ticker_count.get(clean_word, 0) + 1
-
-
-# Helper function to determine if Yahoo Finance can provide the relevant information
-# If not, we skip this stock
-def has_all_finance_fields(stock_info):
-    for field in yahoo_finance_FIELDS:
-        if field not in stock_info.keys():
-            return False
-
-    return True
-
-
-# Helper function to get the most recent Top 10 stock record from the database
-# Returns a list of ticker symbols, sorted by rank
-def get_prev_stock_list(collection, current_timestamp):
-    latest_timestamp_entry = collection.find({'timeStamp': {'$lte': current_timestamp}}).sort([('timeStamp', -1)]).limit(1)
-
-    latest_time_stamp = latest_timestamp_entry[0].get('timeStamp')
-    latest_stock_obj_list = collection.find({'timeStamp': latest_time_stamp})
-
-    sorted_stock_list = sorted(latest_stock_obj_list, key=lambda i: i['rank'])
-
-    latest_stock_list = []
-    for stock_obj in sorted_stock_list:
-        latest_stock_list.append(stock_obj.get('ticker'))
-
-    return latest_stock_list
-
-
-# Helper function to calculate ranking changes for the current Top 10 stocks
-# returns a tuple of whether this stock is new on the list and the ranking change (if applicable)
-def get_ranking_change(prev_stock_list, ticker, rank):
-    is_new = False
-    ranking_change = 0
-
-    if ticker not in prev_stock_list:
-        is_new = True
-        return is_new, ranking_change
-
-    ranking_change = rank - prev_stock_list.index(ticker)
-    return is_new, ranking_change
-
-
-# Packages the Top 10 tickers and related financial information into a list of dictionaries
-# also inserts this information into the database with time stamps
-def insert_to_db(sorted_ticker_dict, ticker_dict):
-    client = MongoClient(os.environ['DB_URL'])
-    db_collection = client['findMyGME']['stocks']
-
-    now = datetime.now()
-
-    to_DB = []
-
-    sorted_ticker_list = list(sorted_ticker_dict.keys())
-
-    prev_stock_list = get_prev_stock_list(db_collection, now)
-
-    rank = 1
-
-    for i in range(TOTAL_NUMBER_OF_STOCKS):
```

Comment:
you should iterate len(sorted_ticker_list), what if sorted_ticker_list has less than 20? you will get index out of bound